extensions/amp-list/0.1/amp-list.js

```diff
@@ -124,21 +133,49 @@ export class AmpList extends AMP.BaseElement {
    * Request list data from `src` and return a promise that resolves when
    * the list has been populated with rendered list items.
    * @return {!Promise}
+   * @private
    */
-  populateList_() {
+  fetchList_() {
     const itemsExpr = this.element.getAttribute('items') || 'items';
-    return this.fetchItems_(itemsExpr).then(items => {
+    return this.fetch_(itemsExpr).then(items => {
       user().assert(isArray(items),
           'Response must contain an array at "%s". %s',
           itemsExpr, this.element);
-      return Services.templatesFor(this.win).findAndRenderTemplateArray(
-          this.element, items).then(this.rendered_.bind(this));
+      return this.renderItems_(items);
     }, error => {
       throw user().createError('Error fetching amp-list', error);
     });
   }
 
   /**
+   * @param {!Array} items
+   * @return {!Promise}
+   * @private
+   */
+  renderItems_(items) {
+    return this.templates_.findAndRenderTemplateArray(this.element, items)
+        .then(this.boundScanForBindings_)
+        .then(this.boundRendered_);
+  }
+
+  /**
+   * @param {!Array<!Element>} elements
+   * @return {!Promise<!Array<!Element>>}
+   * @private
+   */
+  scanForBindings_(elements) {
+    const forwardElements = () => elements;
+    return Services.bindForDocOrNull(this.element).then(bind => {
+      if (bind) {
+        return bind.rescanAndEvaluate(elements);
+      } else {
+        return Promise.resolve();
+      }
+    // Forward elements to chained promise on success or failure.
+    }).then(forwardElements, forwardElements);
```

Comment:
Time for [`Promise#finally`](https://www.npmjs.com/package/promise.prototype.finally)?